Buttons

Woolfe-Glitch opened the pull request:
Updated and Reordered Buttons

garrettdigital reviewed (changes requested):
JF and I reviewed this today. Here are our notes:

- Lost the variant type information (Default, Outlined, Text, etc.)
- States aren't clear (eg: disabled states)

Looking at other design system examples, they change the text in the button to describe the state and/or variant type of the button. We think we should do this as well as it makes it pretty clear as to what you're looking at.
Eg: https://mui.com/material-ui/react-button/, https://ux.mailchimp.com/patterns/buttons, https://getbootstrap.com/docs/5.1/components/buttons/

We were also thinking that we should change the groupings to be context based. Maybe add a one or two sentence blurb for each grouping to describe the correct usage and have that followed by the button examples with the button is a "box" to visually separate them from the content.

### Button Styles
Default | Outlined | Text

### Disable States
Default | Outlined | Text

### Colours
Default | Primary Accent | Secondary
Outlined | Primary Accent | Secondary
Text | Primary Accent | Secondary

### Sizes
Default Large | Default | Small
Outlined Large | Default | Small
Text Large | Default | Small

garrettdigital commented:
> JF and I reviewed this today. Here are our notes:
> 
>     * Lost the variant type information (Default, Outlined, Text, etc.)
> 
>     * States aren't clear (eg: disabled states)
> 
> 
> Looking at other design system examples, they change the text in the button to describe the state and/or variant type of the button. We think we should do this as well as it makes it pretty clear as to what you're looking at. Eg: https://mui.com/material-ui/react-button/, https://ux.mailchimp.com/patterns/buttons, https://getbootstrap.com/docs/5.1/components/buttons/
> 
> We were also thinking that we should change the groupings to be context based. Maybe add a one or two sentence blurb for each grouping to describe the correct usage and have that followed by the button examples with the button is a "box" to visually separate them from the content.
> ### Button Styles
> 
> Default | Outlined | Text
> ### Disable States
> 
> Default | Outlined | Text
> ### Colours
> 
> Default | Primary Accent | Secondary Outlined | Primary Accent | Secondary Text | Primary Accent | Secondary
> ### Sizes
> 
> Default Large | Default | Small Outlined Large | Default | Small Text Large | Default | Small

1. The description text should be more concise. As suggested before, 1 or 2 sentences maximum should be enough to describe a button type. See the linked examples above.
2. Can you please remove the word "Button" from the button labels? It's a bit repetitive having them on most of the buttons.
3. The word "Colour" should use the American spelling "Color".
4. The <hr> surrounding the buttons makes it feel like the buttons are in different sections when they are actually related to/a child of the title and description above them. Is there another way to visually separate the buttons but maintain the information hierarchy with their heading and description?

Woolfe-Glitch commented:
Updated the button writeups. I think the 'general' writeup can't be much shorter, since it describes all the different button states. 

I removed the Canadian wording for Colour, and the extra HR rules to try and pull things a little closer together.
